### DISCUSSION TITLE
posts/pikvm-tesmart/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#304] [Comments]
# posts/pikvm-tesmart/

After many hours and testing, swapping, resetting, and EDID training, all of my PiKVM and TESmart issues were solved with with a simple, cheap dongle. If you aren’t aware of the struggles I faced when using the PiKVM with a TESmart switch, all of it is detailed in a previous post where I ended up settling on another switch altogether. That’s all changed now with a simple EDID emulator passthrough adapter (affiliate link). However, for the uninformed, I’ll summarize the symptoms I experienced with the PiKVM and TESmart KVM and how I was able to fix these issues.

https://technotim.live/posts/pikvm-tesmart/

**Comment by txyaloo** [comment]
This is a "known issue" for them.  Strange they sent replacement hardware.  I went through this same dance with them a few months ago.

My fix  was an HDMI splitter connected to the Tesmart output then PiKVM input.  Cost ~$20 on Amazon.  No need for an individual device at each input.  The splitter I think strips out HDCP and does some sort of edid emulation.  There are a few other benefits to this approach:

1) You can still run a KMM in your rack connected to your KVM by using the second output on the splitter.  
2) You can use the 2nd output into an ethernet HDMI extender.   This gives you access to your entire rack from your living room TV including full keyboard/mouse control at whatever resolution your HDMI extender supports and limited lag since it's not going through the PiKVM.  You can also run the 2nd output into another HDMI splitter if you wanted remote "local" access in other areas.

**Reply by timothystewart6** [reply]
Thanks for the tip!  Yeah, that would have saved me a ton of time!  Thank you for the other tips!  I will check out the splitter!  Can you recommend one?

**Reply by timothystewart6** [reply]
Are you thinking something like this? https://amzn.to/3UD6isq (affiliate link)

**Comment by txyaloo** [comment]
I purchased this one after reading success others had.  Had to do the monitor/unplug/reboot/turn on edid dance, but it works fine since.  Biggest issue I've found is being unable to see the boot screens on a Mac to get into recovery.  It's just analog snow.  Linux/Windows boxes take 1-2 seconds to initialize video after switching inputs, but after that, it's been rock solid.

https://www.amazon.com/gp/product/B082RJY1DB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1

First tried an Onn from walmart but it didn't seem to do the needed HDCP/EDID changes.  Tesmart has some videos on their website showing how the HDMI extenders will work with a splitter if you're interested in that.

**Reply by timothystewart6** [reply]
Thanks for the tips!

**Comment by Fulg** [comment]
After reading your findings I recently bought a TESmart KVM 8x1 4k30 switch and some EDID emulators, and I had the hardest time getting everything to work (the only video mode streamable was 640x480, everything else was broken). Then out of desperation I removed the EDID emulators, and lo and behold everything worked.

I think the issue is that by putting a 4k EDID emulator between the switch and a PC, that PC will think 4k is supported (that's what the emulator does after all) so it will use it if it can, which breaks PIKVM streaming. I'm surprised you didn't see this behavior on your side with your 4k EDID emulators.

If I use a 4k EDID emulator on a PC that _does not_ support 4k output then it works fine as well. So it looks like it is important to get FHD/1080p EDID emulators to avoid breaking PIKVM itself... or stick to hardware that can't output 4k :)

Thanks for posting this, I've long wanted to control/troubleshoot the multiple PCs in my rack with a single PIKVM, and now I can!

**Comment by ivan-zaitsev** [comment]
Is the TESmart Switch works both with PiKVM Mini and Plus, or only Plus?

**Comment by iLLiac4** [comment]
https://mysku.club/blog/aliexpress/101591.html Use translate and read some more EDID deep dive into modifying emulators.
